react/features/filmstrip/components/web/Filmstrip.js

```diff
@@ -306,37 +433,112 @@ class Filmstrip extends PureComponent <Props> {
+                                    key = 'local' />
+                            }
+                        </div>
+                    </div>
+                )}
+                {
+                    this._renderRemoteParticipants()
+                }
+            </div>
+            <AudioTracksContainer />
+        </>);
+
         return (
             <div
                 className = { clsx('filmstrip',
                     this.props._className,
-                    classes.filmstrip) }
+                    classes.filmstrip,
+                    _verticalFilmstripWidth + FILMSTRIP_BREAKPOINT_OFFSET >= FILMSTRIP_BREAKPOINT
+                        && classes.filmstripBackground) }
+                ref = { this._filmstripRef }
                 style = { filmstripStyle }>
                 { toolbar }
-                <div
-                    className = { this.props._videosClassName }
-                    id = 'remoteVideos'>
-                    {!_disableSelfView && (
+                {_resizableFilmstrip
+                    ? <div className = { classes.filmstripContainer }>
                         <div
-                            className = 'filmstrip__videos'
-                            id = 'filmstripLocalVideo'>
-                            <div id = 'filmstripLocalVideoThumbnail'>
-                                {
-                                    !tileViewActive && <Thumbnail
-                                        key = 'local' />
-                                }
-                            </div>
-                        </div>
-                    )}
-                    {
-                        this._renderRemoteParticipants()
-                    }
-                </div>
-                <AudioTracksContainer />
+                            className = { clsx('dragHandle',
+                                classes.dragHandle,
+                                isMouseDown && 'show')
+                            }
+                            onMouseDown = { this._onDragHandleMouseDown } />
+                        {filmstrip}
+                    </div>
+                    : filmstrip
+                }
             </div>
         );
     }
 
+    _onDragHandleMouseDown: (MouseEvent) => void;
+
+    /**
+     * Handles mouse down on the drag handle.
+     *
+     * @param {MouseEvent} e - The mouse down event.
+     * @returns {void}
+     */
+    _onDragHandleMouseDown(e) {
+        this._dragRef.current.down = true;
+        this._dragRef.current.initialPosition = e.clientX;
+        this._dragRef.current.initialWidth = this.props._verticalFilmstripWidth || DEFAULT_FILMSTRIP_WIDTH;
+        this.setState({
+            isMouseDown: true
+        });
+    }
+
+    _onDragMouseUp: () => void;
+
+    /**
+     * Drag handle mouse up handler.
+     *
+     * @returns {void}
+     */
+    _onDragMouseUp() {
+        if (this._dragRef.current.down) {
+            this._dragRef.current.down = false;
```

Comment:
nit . do we need 2 props for mouseDown bool? We could only keep it in the state prop